Sources/SKSwiftPMWorkspace/SwiftPMBuildSystem.swift

```diff
@@ -368,6 +391,35 @@ extension SwiftPMBuildSystem: SKCore.BuildSystem {
         try await self.reloadPackage()
       }
     }
+
+    var filesWithUpdatedDependencies: Set<DocumentURI> = []
+    // If a Swift file within a target is updated, reload all the other files within the target since they might be
+    // referring to a function in the updated file.
+    for event in events {
+      guard let url = event.uri.fileURL,
+        url.pathExtension == "swift",
+        let absolutePath = try? AbsolutePath(validating: url.path),
+        let target = fileToTarget[absolutePath]
+      else {
+        continue
+      }
+      filesWithUpdatedDependencies.formUnion(target.sources.map { DocumentURI($0) })
+    }
+
+    // If a `.swiftmodule` file is updated, this means that we have performed a build / are
+    // performing a build and files that depend on this module have updated dependencies.
+    // We don't have access to the build graph from the SwiftPM API offered to SourceKit-LSP to figure out which files
+    // depend on the updated module, so assume that all files have updated dependencies. Currently, this does not matter
+    // because `SwiftLanguageService` sends a `dependencyUpdated` request to sourcekitd, which is global anyway.
```

Comment:
FWIW that only seems to impact completion, not eg. cursor info.

Reply:
Ah, we do also reopen the file. I updated the comment to just remove the

> Currently, this does not matter because `SwiftLanguageService` sends a `dependencyUpdated` request to sourcekitd, which is global anyway.